internal/services/springcloud/spring_cloud_app_resource.go

```diff
@@ -560,7 +560,7 @@ func flattenAppCustomPersistentDiskResourceArray(input *[]appplatform.CustomPers
 	for _, item := range *input {
 		var storageName string
 		if item.StorageID != nil {
-			if id, err := parse.SpringCloudStorageID(*item.StorageID); err == nil {
+			if id, err := parse.SpringCloudStorageIDInsensitively(*item.StorageID); err == nil {
```

Comment:
This indicates that the service is storing, or just returning, the value in the incorrect casing. Can we get an issue opened with the service team and noted here for tracking?